pandas/core/internals/managers.py

```diff
@@ -596,18 +596,24 @@ def replace_list(
         # figure out our mask apriori to avoid repeated replacements
         values = self.as_array()
 
-        def comp(s, regex=False):
+        def comp(s, regex=False, mask=None):
             """
             Generate a bool array by perform an equality check, or perform
             an element-wise regular expression matching
             """
             if isna(s):
-                return isna(values)
+                if mask is None:
+                    return isna(values)
```

Comment:
yeah this looks wrong actually so i bet it is not called?

Reply:
Yes, since mask is now required this is not useful anymore. Removed.